test/generateDefinition.test.ts

```diff
@@ -105,9 +105,56 @@ const {temperature, onOff} = require('zigbee-herdsman-converters/lib/modernExten
 const definition = {
     zigbeeModel: ['combo'],
     model: 'combo',
-    vendor: 'undefined',
+    vendor: 'vendor',
     description: 'Automatically generated definition',
-    extend: [temperature(), onOff({powerOnBehavior: false})],
+    extend: [temperature({"endpoints":["1"]}), onOff({"powerOnBehavior":false})],
+    meta: {},
+};
+
+module.exports = definition;
+            `
+        });
+    });
+
+    test('input(msTemperatureMeasurement, genOnOff, msTemperatureMeasurement)', async () => {
+        await assertGeneratedDefinition({
+            device: mockDevice({modelID: 'combo', endpoints: [
+                {inputClusters: ['msTemperatureMeasurement', 'genOnOff'], outputClusters:[]},
+                {ID: 2, inputClusters: ['msTemperatureMeasurement'], outputClusters: []},
+            ]}),
+            meta: {multiEndpoint: true},
+            fromZigbee: [expect.objectContaining({cluster: 'msTemperatureMeasurement'}), fz.on_off],
+            toZigbee: ['temperature', 'state', 'on_time', 'off_wait_time'],
+            exposes: ['linkquality', 'switch(state)', 'temperature', 'temperature'],
+            bind: {1: ['msTemperatureMeasurement', 'genOnOff'], 2: ['msTemperatureMeasurement']},
+            read: {
+                1: [
+                    ['msTemperatureMeasurement', ['measuredValue']],
+                    ['genOnOff', ['onOff']],
+                ],
+                2: [
+                    ['msTemperatureMeasurement', ['measuredValue']],
+                ]
+            },
+            configureReporting: {
+                1: [
+                    ['msTemperatureMeasurement', [reportingItem('measuredValue', 10, repInterval.HOUR, 100)]],
+                    ['genOnOff', [reportingItem('onOff', 0, repInterval.MAX, 1)]],
+                ],
+                2: [
+                    ['msTemperatureMeasurement', [reportingItem('measuredValue', 10, repInterval.HOUR, 100)]],
+                ],
+            },
+            externalDefintionSource: `
+const {temperature, onOff} = require('zigbee-herdsman-converters/lib/modernExtend');
+
+const definition = {
+    zigbeeModel: ['combo'],
+    model: 'combo',
+    vendor: '',
+    description: 'Automatically generated definition',
+    extend: [temperature({"endpoints":["1","2"]}), onOff({"powerOnBehavior":false})],
```

Comment:
For other modernExtend the endpoints passed is an object, e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/e4b88b8704bc6baa00671a685eec0ce12c6781b0/src/devices/atsmart.ts#L10

This allows to specify a friendly name for the endpoint

Reply:
Honestly I think that in this particular example the way that `onOff` handles some of its argument is "excessive".
I would propose to split it in to more extenders, like `[onOff({endpoints: ["left","center","right"]}), deviceEndpoints({left:1, center:2, right:3}), ota(someDefinition)]`, as I think it should not define which endpoints device has, nor, even more so, define the OTA configuration.

What do you think?

Reply:
Updated. I am not sure what specific check to do, so I didn't do any. If you would suggest which checks to do - I can give them a go.

I also didn't update current usages of `onOff` with endpoints, as it would increase the size of this MR. I can do it in the next MR, if this one will be merged.

Reply:
Sounds good! Already merged this to get it into the 1 Feb release.